readthedocs/api/v2/views/model_views.py

```diff
@@ -345,6 +348,36 @@ def reset(self, request, **kwargs):
     def get_queryset_for_api_key(self, api_key):
         return self.model.objects.filter(project=api_key.project)
 
+    @decorators.action(
+        detail=True,
+        permission_classes=[HasBuildAPIKey],
+        methods=["post"],
+        url_path="temporary-credentials",
+    )
+    def temporary_credentials(self, request, **kwargs):
+        """
+        Generate temporary credentials for the build.
+
+        This can generate temporary credentials for interacting with S3 only for now.
+        """
+        build = self.get_object()
+        project = build.project
+        version = build.version
+        try:
+            credentials = get_s3_scoped_credentials(
+                project=project,
+                version=version,
+                session_id=build.pk,
+                # 30 minutes should be enough to upload all build artifacts.
+                duration=30 * 60,
```

Comment:
We have builds that are configured longer than this. We should at least do `project.container_time_limit * 2` at least. Is there a strong reason to be conservative here, if it's only accessing a single bucket directory?

Reply:
The credentials are created just before they are needed, that's it, just before the upload step, so there is no need for them to live as long as the whole build.